src/runmode-dpdk.c

```diff
@@ -0,0 +1,1274 @@
+        }
+    }
+    SCReturnInt(0);
+}
+
+int DeviceConfigure(DPDKIfaceConfig *iconf)
+{
+    // configure device
+    int retval;
+    struct rte_eth_dev_info dev_info;
+    struct rte_eth_conf port_conf;
+
+    retval = rte_eth_dev_get_port_by_name(iconf->iface, &(iconf->port_id));
+    if (retval < 0) {
+        SCLogError(SC_ERR_DPDK_INIT, "Error (err=%d) when getting port id of %s Is device enabled?",
+                retval, iconf->iface);
+        SCReturnInt(retval);
+    }
+
+    if (!rte_eth_dev_is_valid_port(iconf->port_id)) {
+        SCLogError(SC_ERR_DPDK_INIT, "Specified port %d is invalid", iconf->port_id);
+        SCReturnInt(retval);
+    }
+
+    retval = rte_eth_dev_socket_id(iconf->port_id);
+    if (retval < 0) {
+        SCLogError(SC_ERR_DPDK_INIT, "Error (err=%d) invalid socket id (port %u)", retval,
+                iconf->port_id);
+        SCReturnInt(retval);
+    } else {
+        iconf->socket_id = retval;
+    }
+
+    retval = rte_eth_dev_info_get(iconf->port_id, &dev_info);
+    if (retval != 0) {
+        SCLogError(SC_ERR_DPDK_INIT, "Error (err=%d) during getting device info (port %u)", retval,
+                iconf->port_id);
+        SCReturnInt(retval);
+    }
+
+    DeviceInitPortConf(iconf, &dev_info, &port_conf);
+    if (port_conf.rxmode.offloads & DEV_RX_OFFLOAD_CHECKSUM) {
+        // Suricata does not need recalc checksums now
+        iconf->checksum_mode = CHECKSUM_VALIDATION_DISABLE;
+    }
+
+    retval = rte_eth_dev_configure(
+            iconf->port_id, iconf->nb_rx_queues, iconf->nb_tx_queues, &port_conf);
+    if (retval != 0) {
+        SCLogError(SC_ERR_DPDK_INIT, "Error (err=%d) during configuring the device (port %u)",
+                retval, iconf->port_id);
+        SCReturnInt(retval);
+    }
+
+    retval = rte_eth_dev_adjust_nb_rx_tx_desc(
+            iconf->port_id, &iconf->nb_rx_desc, &iconf->nb_tx_desc);
+    if (retval != 0) {
+        SCLogError(SC_ERR_DPDK_INIT,
+                "Error (err=%d) during adjustment of device queues descriptors (port %u)", retval,
+                iconf->port_id);
+        SCReturnInt(retval);
+    }
+
+    retval = iconf->flags & DPDK_MULTICAST ? rte_eth_allmulticast_enable(iconf->port_id)
+                                           : rte_eth_allmulticast_disable(iconf->port_id);
+    if (retval < 0) {
+        SCLogError(SC_ERR_DPDK_INIT, "Error (err=%d) when en/disabling multicast on port %u",
+                retval, iconf->port_id);
+        SCReturnInt(retval);
+    }
+
+    // set maximum transmission unit
+    SCLogConfig("Setting MTU of %s to %dB", iconf->iface, iconf->mtu);
+    retval = rte_eth_dev_set_mtu(iconf->port_id, iconf->mtu);
+    if (retval < 0) {
+        SCLogError(SC_ERR_DPDK_INIT, "Error (err=%d) when setting MTU to %u on port %u", retval,
+                iconf->mtu, iconf->port_id);
+        SCReturnInt(retval);
+    }
+
+    retval = DeviceConfigureQueues(iconf, &dev_info, &port_conf);
```

Comment:
please explain or at least comment, why one need to invoke DeviceConfigureQueues after a single DPDK port in configured and initialized.

Reply:
cure code shows you were creating a separate mempool per device on the Numa where lcore is located. This drastically affects the NUMA based huge page especially DPDK message allocation. I highly recommend revisiting and rework.

Reply:
I've changed it to create a separate mempool per device on the NUMA where the NIC is located. 
Thanks. But overall, it is highly recommended to have the whole Suricata setup on 1 NUMA.